Firmware/ultralcd.cpp

```diff
@@ -5297,81 +5299,87 @@ static void lcd_main_menu()
 #endif
 
     // only allow starting SD print if hardware errors (temperature or fan) are cleared
-    if(!get_temp_error()
-            && !printer_active()
+    if (!printer_recovering() && !printer_active() && !get_temp_error()
```

Comment:
Potential optimisation: there are a few if statements inside this top-level if statements which check for `!printer_active()`. That shouldn't be needed anymore since this top-level if statement checks it.

Reply:
Right, thanks